.circleci/config.yml

```
@@ -453,6 +453,14 @@ jobs:
             RUN_ENV=tests venv/bin/flask install_data
             RUN_ENV=tests venv/bin/python tests/alembic/check_db_schema.py
           working_directory: api
+      - run:
+          name: Check that downgrade scripts are correctly written
+          command: |
+            RUN_ENV=tests venv/bin/alembic downgrade pre@head-10
+            RUN_ENV=tests venv/bin/alembic upgrade pre@head
+            RUN_ENV=tests venv/bin/alembic downgrade post@head-10
+            RUN_ENV=tests venv/bin/alembic upgrade post@head
```

pourquoi ne pas faire les 2 downgrades ensembles ?

```suggestion
            RUN_ENV=tests venv/bin/alembic downgrade post@head-10
            RUN_ENV=tests venv/bin/alembic downgrade pre@head-10
            RUN_ENV=tests venv/bin/alembic upgrade pre@head
            RUN_ENV=tests venv/bin/alembic upgrade post@head
```

---

L'ordre ne me semble pas avoir d'importance, j'ai procédé branche par branche.

---

j'avais cru comprendre qu'il y avait un `pre` et un `post` pour gérer les cas commme l'ajout d'une colonne non nulle

* `pre` : on ajoute la colonne nullable puis on remplit avec une valeur par défaut
* `post` : on change la colonne pour qu'elle ne soit pas nullable

dans ce cas, il me semble que l'ordre à une importance, non ?

ai-je mal compris l'utilité des `pre` et `post` ?

---

Oui, tu as raison, la branche `post` ne devrait pas être downgradée avant la branche `pre`, ta syntaxe est plus sûre.

---

Oui, tu as raison, il faut d'abord _downgrade_ la branche post !
En testant, je me suis rendu compte que l'on utilisait une très vieille version d'`alembic` qui contenait un bug de résolution de révision relative, donc j'en profite pour _bump_ `alembic`.